### ISSUE TITLE
ZeroDivisionError: division by zero

### ISSUE THREAD

**Issue by aliechoes** [epigenelabs/pyComBat#16]
Hi

thanks for the library and your work. 
I am facing a weird problem. I am using the `pycombat` function. However this is the error that I get repeatedly. 

```
ZeroDivisionError                         Traceback (most recent call last)
<ipython-input-119-bb601998f8f1> in <module>
      1 # run pyComBat
----> 2 df_corrected = pycombat(df_features_compounds.T, 
      3                         metadata["Donor"].tolist())
      4 
      5 

~/.conda/envs/monai/lib/python3.8/site-packages/combat/pycombat.py in pycombat(data, batch, mod, par_prior, prior_plots, mean_only, ref_batch, precision, **kwargs)
    655     NAs = check_NAs(dat)
    656     if not(NAs):
--> 657         B_hat, grand_mean, var_pooled = calculate_mean_var(
    658             design, batches, ref, dat, NAs, ref_batch, n_batches, n_batch, n_array)
    659         stand_mean = calculate_stand_mean(

~/.conda/envs/monai/lib/python3.8/site-packages/combat/pycombat.py in calculate_mean_var(design, batches, ref, dat, NAs, ref_batch, n_batches, n_batch, n_array)
    451     else:
    452         grand_mean = np.dot(np.transpose(
--> 453             [i / n_array for i in n_batches]), B_hat[0:n_batch])
    454     # Calculates the general variance
    455     if not NAs:  # NAs not supported

~/.conda/envs/monai/lib/python3.8/site-packages/combat/pycombat.py in <listcomp>(.0)
    451     else:
    452         grand_mean = np.dot(np.transpose(
--> 453             [i / n_array for i in n_batches]), B_hat[0:n_batch])
    454     # Calculates the general variance
    455     if not NAs:  # NAs not supported

ZeroDivisionError: division by zero

```

I exactly don't know what to do about it. Would you please help me?

**Comment by abdelkaderbehdenna** [comment]
Hi,
Thank you for using pyComBat, and for your feedback. It is very helpful for us to receive such interactions.
Unfortunately, it would be challenging for me to reproduce the aforementioned issue.
Could you please provide a simple (or dummy) data and a snippet of the code that leads to this error? In the meantime, it may be worth your time to investigate the presence of NAs or zeros in your data.

Best,
Kader

**Comment by aliechoes** [comment]
Hey Kader. 

Thanks for the response. I actually could find the problem. in my code I had 

```python
df_corrected = pycombat(df_features_compounds.T,  metadata["Donor"].tolist())
```

I noticed that I should not use `list` for the batch but a `pd.Series` instead. Maybe it could make sense to add this in the documentation.

Thanks again for the libray
Cheers

**Comment by EmilyMiddendorp** [comment]
> Hey Kader.
> 
> Thanks for the response. I actually could find the problem. in my code I had
> 
> ```python
> df_corrected = pycombat(df_features_compounds.T,  metadata["Donor"].tolist())
> ```
> 
> I noticed that I should not use `list` for the batch but a `pd.Series` instead. Maybe it could make sense to add this in the documentation.
> 
> Thanks again for the libray
> Cheers

I just got the same exact error and this solution worked for me as well!  Thank you!

**Comment by moldovannorbert** [comment]
> Hey Kader.
> 
> Thanks for the response. I actually could find the problem. in my code I had
> 
> ```python
> df_corrected = pycombat(df_features_compounds.T,  metadata["Donor"].tolist())
> ```
> 
> I noticed that I should not use `list` for the batch but a `pd.Series` instead. Maybe it could make sense to add this in the documentation.
> 
> Thanks again for the libray Cheers

Same issue same solution! The documentation needs to be corrected.

**Comment by abdelkaderbehdenna** [comment]
Hey all,

Sorry for the delay, we added a control test within the code of pycombat so that this problem wouldn't happen again.

For information, this was related to batches defined as a list of string that would cause an issue with np.where. It is now corrected and tested in PR #25 .

Thank you all again for notifying us :)

Kader
